crates/re_chunk_store/src/query.rs

```diff
@@ -0,0 +1,236 @@
+            .get(entity_path)
+            .and_then(|temporal_chunk_ids_per_timeline| {
+                temporal_chunk_ids_per_timeline.get(&query.timeline())
+            })
+            .and_then(|temporal_chunk_ids_per_component| {
+                temporal_chunk_ids_per_component.get(&component_name)
+            })
+            .and_then(|temporal_chunk_ids_per_time| {
+                temporal_chunk_ids_per_time
+                    .per_start_time
+                    .range(..=query.at())
+                    .next_back()
+                    .map(|(_time, chunk_ids)| chunk_ids)
+            })
+        {
+            return temporal_chunk_ids
+                .iter()
+                .filter_map(|chunk_id| self.chunks_per_chunk_id.get(chunk_id).cloned())
+                .collect();
+        }
+
+        Vec::new()
+    }
+
+    /// Returns the most-relevant chunk(s) for the given [`RangeQuery`].
+    ///
+    /// It is possible that some or all of the returned [`Chunk`]s overlap in time.
+    ///
+    /// The caller should filter the returned chunks further (see [`Chunk::range`]) in order to
+    /// determine how exactly each row of data fit with the rest.
+    ///
+    /// If the entity has static component data associated with it, it will unconditionally
+    /// override any temporal component data.
+    pub fn range(
+        &self,
+        query: &RangeQuery,
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+    ) -> Vec<Arc<Chunk>> {
+        re_tracing::profile_function!(format!("{query:?}"));
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        if let Some(static_chunk) = self
+            .static_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|static_chunks_per_component| {
+                static_chunks_per_component.get(&component_name)
+            })
+            .and_then(|chunk_id| self.chunks_per_chunk_id.get(chunk_id))
+        {
+            return vec![Arc::clone(static_chunk)];
+        }
+
+        self.temporal_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|temporal_chunk_ids_per_timeline| {
+                temporal_chunk_ids_per_timeline.get(&query.timeline())
+            })
+            .and_then(|temporal_chunk_ids_per_component| {
+                temporal_chunk_ids_per_component.get(&component_name)
+            })
+            .into_iter()
+            .map(|temporal_chunk_ids_per_time| {
+                let start_time = temporal_chunk_ids_per_time
+                    .per_start_time
+                    .range(..=query.range.min())
+                    .next_back()
+                    .map_or(TimeInt::MIN, |(&time, _)| time);
+
+                let end_time = temporal_chunk_ids_per_time
+                    .per_start_time
+                    .range(..=query.range.max())
+                    .next_back()
+                    .map_or(start_time, |(&time, _)| time);
```

Comment:
Ditto here; this looks very obviously wrong for cases where a chunk completely overlaps another one? write a test and fix this code.